### PR TITLE
Apply clang-format to stereo module

### PR DESCRIPTION
**Pull request by taketwo**

### CODE REVIEW

File: stereo/src/stereo_adaptive_cost_so.cpp [hunk 1]
```diff
@@ -36,167 +36,188 @@
...
  */
 
-/** \brief Adaptive Cost 2-pass Scanline Optimization Stereo Matching algorithm implementation
-  * please see related documentation on stereo/stereo_matching.h
-  *
-  * \author Federico Tombari (federico.tombari@unibo.it)
-  * \ingroup stereo
-  */
+/** \brief Adaptive Cost 2-pass Scanline Optimization Stereo Matching algorithm
+ * implementation please see related documentation on stereo/stereo_matching.h
```
Comment (kunaltyagi) [inline stereo/src/stereo_adaptive_cost_so.cpp]: Looks weird. Also doxygen has related functionality [`@see`](http://www.doxygen.nl/manual/autolink.html)

File: stereo/src/stereo_adaptive_cost_so.cpp [hunk 2]
```diff
@@ -36,167 +36,188 @@
...
-    }//x
- 
-    //last scan
-    for (int x = x_off_ + max_disp_ + 1; x < width_; x++)
-    {
-      float c_min = std::numeric_limits<float>::max ();
+        if (bck[x + 1][d] < c_min)
+          c_min = bck[x + 1][d];
+
+      bck[x][0] =
```
Comment (kunaltyagi) [inline stereo/src/stereo_adaptive_cost_so.cpp]: Might need to refactor this to be more understandable later.

File: stereo/src/stereo_adaptive_cost_so.cpp [hunk 3]
```diff
@@ -36,167 +36,188 @@
...
-  //int n = radius_ * 2 + 1;
-  //int sad_max = std::numeric_limits<int>::max ();
- 
-  float **acc = new float *[width_];
-  for (int d = 0; d < width_; d++)
-  {
+  // int n = radius_ * 2 + 1;
+  // int sad_max = std::numeric_limits<int>::max ();
+
+  float** acc = new float*[width_];
```
Comment (kunaltyagi) [inline stereo/src/stereo_adaptive_cost_so.cpp]: Can we use some smarter memory management? Even `std::vector<std::vector<float>>` might help. There might be some 2D data-struct in PCL that I don't know about.

Same with the `fwd` and `bck` arrays. The code works, but seems fragile

File: stereo/src/stereo_adaptive_cost_so.cpp (hunk 1 above)
Comment (taketwo) [reply stereo/src/stereo_adaptive_cost_so.cpp]: Hm, indeed this docstring has no effect I think, at least I can not find it in the generated documentation. Will wipe it out.

File: stereo/src/stereo_adaptive_cost_so.cpp (hunk 3 above)
Comment (taketwo) [reply stereo/src/stereo_adaptive_cost_so.cpp]: I would absolutely not admit any new code with such memory management. But I'm very hesitant to touch this... I have never used `stereo` module and also I don't know how comprehensive its test coverage is. For me the risks of breaking out-weight the benefits.

File: stereo/src/stereo_adaptive_cost_so.cpp (hunk 2 above)
Comment (taketwo) [reply stereo/src/stereo_adaptive_cost_so.cpp]: Same here :)

File: stereo/src/stereo_adaptive_cost_so.cpp (hunk 3 above)
Comment (kunaltyagi) [reply stereo/src/stereo_adaptive_cost_so.cpp]: > 2D data-struct in PCL that I don't know about

Does one exist? What do other modules use? `vector<vector<T>>` is not really a 2D matrix (nor is T**).

> I don't know how comprehensive its test coverage is

Where are the tests? I didn't see a `test/stereo` directory

Comment (taketwo) [reply stereo/src/stereo_adaptive_cost_so.cpp]: > Does one exist? What do other modules use?

I think `Eigen::MatrixXf` would be an appropriate choice here.

> Where are the tests? I didn't see a `test/stereo` directory. 

Indeed, no test coverage at all. Unfortunately PCL has a number of such dark dusty corners.

> I'll create some tests and refactor this if it's ok?

Any contribution is welcome! That said, if you are willing to volunteer time, I believe there are more important things in PCL that need someone's attention.

**Comment by taketwo** [comment]
Removed doxygen comments from the top of ".cpp" files, everything else is the same.

**Pull request merged**